pallets/compliance-manager/src/lib.rs

```diff
@@ -656,14 +699,24 @@ impl<T: Config> ComplianceManagerConfig for Module<T> {
 
         for req in asset_compliance.requirements {
             if let Some(from_did) = from_did_opt {
-                if !Self::are_all_conditions_satisfied(ticker, from_did, &req.sender_conditions) {
```

Comment:
We might need to charge a weight for the cost of this loop and the loop in `are_all_conditions_satisfied`.  It most likely isn't much, but we need to make sure the overhead isn't too much.

Reply:
I added a benchmark for this. Here is the cost of the loops: 
```Rust
Weight::from_ref_time(1_151_180)
    // Standard Error: 1_742
    .saturating_add(Weight::from_ref_time(71_878).saturating_mul(i.into()))
```     
Where `i` is the complexity of the requirements. The number of iterations for all loops  can be at most `MaxConditionComplexity`, since both the number of requirements and conditions add to the complexity. I was able to remove the if in a different branch.